utils/log/progress.go

```diff
@@ -0,0 +1,11 @@
+package log
+
+import "io"
+
+type ProgressBar interface {
```

Comment:
I suggest renaming *ProgressBar* *Progress*, because people can implement there own progress indicators and not necessarily implement a progress bar.
I also suggest renaming all the *progressBar* variables throughout the code to *progress*.
I'd also suggest adding a description to this interface, explaining that it is used for file transfer progress indication.